Marv-Lous/Marv-Lous/ViewModels/LoginPageViewModel.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  LoginPageViewModel.swift
+//  Marv-Lous
+//
+//  Created by Erik Egers on 2022/04/01.
+//
+
+import Foundation
+
+protocol LoginPageViewModelDelegate: AnyObject {
+    func showError(_ error: String)
+    func navigateToLandingPage()
+}
+
+class LoginPageViewModel {
+    private weak var delegate: LoginPageViewModelDelegate?
+    private var username = ""
+    private var password = ""
+    
+    init(delegate: LoginPageViewModelDelegate?) {
+        self.delegate = delegate
+    }
+    
+    func attemptLogin(_ username: String, _ password: String) {
+        self.username = username
+        self.password = password
+        
+        switch credentialsInput() {
+        case .correct:
+            self.delegate?.navigateToLandingPage()
```

Comment:
So why not call this on line 47 😄 then we can remove the `enum` as well.

Reply:
That makes sense, it is much cleaner now.